core/src/test/scala/org/typelevel/paiges/PaigesScalacheckTest.scala

```diff
@@ -254,6 +235,48 @@ class PaigesScalacheckTest extends FunSuite {
     }
   }
 
+  test("Union invariant: `a.flatten == b.flatten`") {
+    forAll { d: Doc.Union => assert(d.a.flatten === d.b.flatten) }
+  }
+
+  test("Union invariant: `a != b`") {
+    forAll { d: Doc.Union => assert(d.a !== d.b) }
+  }
+
+  def unionRightAssocInvariant(d: Doc): Assertion = {
```

Comment:
I'm going to use this in other tests in a bit, which is why it's not nested in the invariant test.